addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -179,315 +446,795 @@ MockServer.include({
+                },
+            ];
+            this._widget.call('bus_service', 'trigger', 'notification', [notification]);
+        }
+    },
+    /**
+     * Simulates `channel_fetch_preview` on `mail.channel`.
+     *
+     * @private
+     * @param {integer[]} ids
+     * @return {Object[]} list of channels previews
+     */
+    _mockMailChannelChannelFetchPreview(ids) {
+        const channels = this._getRecords('mail.channel', [['id', 'in', ids]]);
+        return channels.map(channel => {
+            const channelMessages = this._getRecords('mail.message', [['channel_ids', 'in', channel.id]]);
+            const lastMessage = channelMessages.reduce((lastMessage, message) => {
+                if (message.id > lastMessage.id) {
+                    return message;
+                }
+                return lastMessage;
+            }, channelMessages[0]);
+            return {
+                id: channel.id,
+                last_message: lastMessage ? this._mockMailMessageMessageFormat([lastMessage.id])[0] : false,
+            };
+        });
+    },
+    /**
+     * Simulates the 'channel_fold' route on `mail.channel`.
+     * In particular sends a notification on the bus.
+     *
+     * @private
+     * @param {string} uuid
+     * @param {state} [state]
+     */
+    _mockMailChannelChannelFold(uuid, state) {
+        const channel = this._getRecords('mail.channel', [['uuid', '=', uuid]])[0];
+        this._mockWrite('mail.channel', [channel.id], {
+            is_minimized: state !== 'closed',
+            state,
+        });
+        const notifConfirmFold = [
+            ["dbName", 'res.partner', this.currentPartnerId],
+            this._mockMailChannelChannelInfo([channel.id])[0]
+        ];
+        this._widget.call('bus_service', 'trigger', 'notification', [notifConfirmFold]);
+    },
+    /**
+     * Simulates 'channel_get' on 'mail.channel'.
+     *
+     * @private
+     * @param {integer[]} [partners_to=[]]
+     * @param {boolean} [pin=true]
+     * @returns {Object}
+     */
+    _mockMailChannelChannelGet(partners_to = [], pin = true) {
+        if (partners_to.length === 0) {
+            return false;
+        }
+        if (!partners_to.includes(this.currentPartnerId)) {
+            partners_to.push(this.currentPartnerId);
+        }
+        const partners = this._getRecords('res.partner', [['id', 'in', partners_to]]);
 
-        // add main suggestions based on users
-        const partnersFromUsers = this.data['res.users'].records
-            .map(user => this.data['res.partner'].records[user.partner_id])
-            .filter(partner => partner);
-        const mainMatchingPartners = mentionSuggestionsFilter(partnersFromUsers, search, limit);
+        // NOTE: the server might return an already existing channel for those
+        // partners if possible. This is currently not done here for simplicity,
+        // because no test is relying on it at the time of this writing.
```

Comment:
changed
        // NOTE: this mock is not complete, which is done for simplicity.
        // Indeed if a chat already exists for the given partners, the server
        // is supposed to return this existing chat. But the mock is currently
        // always creating a new chat, because no test is relying on receiving
        // an existing chat.